chain-impl-mockchain/src/certificate/vote_tally.rs

```diff
@@ -40,8 +40,14 @@ pub enum TallyProof {
 }
 
 #[derive(Debug, Eq, PartialEq, Hash, Clone)]
-pub struct TallyDecryptShares {
-    inner: Box<[Box<[TallyDecryptShare]>]>,
+pub struct PrivateTallyDecrypted {
+    inner: Box<[PrivateTallyDecryptedProposal]>,
+}
+
+#[derive(Debug, Eq, PartialEq, Hash, Clone)]
+pub struct PrivateTallyDecryptedProposal {
+    pub decrypt_shares: Box<[TallyDecryptShare]>,
+    pub tally_result: Box<[u64]>,
```

Comment:
Shouldn't the result be included just once? It does not need to be duplicated per each share.

Reply:
Sorry, confused about the array's dimensions.